frontend/public/module/k8s/k8s.ts

```diff
@@ -46,46 +45,54 @@ const parseAPIVersion = (version: string) => {
     : null;
 };
 
-export const crdVersionSort = (crdVersions: CRDVersion[]) => {
-  return crdVersions?.sort((v1, v2) => {
-    const v1Parsed = parseAPIVersion(v1.name);
-    const v2Parsed = parseAPIVersion(v2.name);
-    // First sort on major version with no qualifiers: v3 > v1
-    if (
-      v1Parsed.majorVersion !== v2Parsed.majorVersion &&
-      !v1Parsed.qualifier &&
-      !v2Parsed.qualifier
-    ) {
-      return v2Parsed.majorVersion - v1Parsed.majorVersion;
-    }
-    // Then sort on any version with no qualifier over a qualifier: v1 > v3alpha
-    if (_.isEmpty(v1Parsed.qualifier) !== _.isEmpty(v2Parsed.qualifier)) {
-      return v1Parsed.qualifier ? 1 : -1;
-    }
-    // Beta beats alpha: v1beta1 > v1alpha1
-    const isBetaV1 = v1Parsed.qualifier === 'beta';
-    const isBetaV2 = v2Parsed.qualifier === 'beta';
-    if (isBetaV1 !== isBetaV2) {
-      return isBetaV1 ? -1 : 1;
-    }
-    // Same qualifier, then numeric values win: v2beta2 > v1beta2
-    if (v1Parsed.majorVersion !== v2Parsed.majorVersion) {
-      return v2Parsed.majorVersion - v1Parsed.majorVersion;
-    }
-    // Finally compare minor version: v1beta2 > v1beta1
-    return v2Parsed.minorVersion - v1Parsed.minorVersion;
-  });
+export const crdVersionSort = (v1: string, v2: string) => {
+  const v1Parsed = parseAPIVersion(v1);
+  const v2Parsed = parseAPIVersion(v2);
+
+  // Check null parsed versions first
+  if (_.isEmpty(v1Parsed) || _.isEmpty(v2Parsed)) {
+    // If a value fails null check order it last
+    return v1Parsed ? -1 : v2Parsed ? 1 : 0;
+  }
+  // Then sort on major version with no qualifiers: v3 > v1
+  if (
+    v1Parsed.majorVersion !== v2Parsed.majorVersion &&
+    !v1Parsed.qualifier &&
+    !v2Parsed.qualifier
+  ) {
+    return v2Parsed.majorVersion - v1Parsed.majorVersion;
+  }
+  // Then sort on any version with no qualifier over a qualifier: v1 > v3alpha
+  if (_.isEmpty(v1Parsed.qualifier) !== _.isEmpty(v2Parsed.qualifier)) {
+    return v1Parsed.qualifier ? 1 : -1;
+  }
+  // Beta beats alpha: v1beta1 > v1alpha1
+  const isBetaV1 = v1Parsed.qualifier === 'beta';
+  const isBetaV2 = v2Parsed.qualifier === 'beta';
+  if (isBetaV1 !== isBetaV2) {
+    return isBetaV1 ? -1 : 1;
+  }
+  // Same qualifier, then numeric values win: v2beta2 > v1beta2
+  if (v1Parsed.majorVersion !== v2Parsed.majorVersion) {
+    return v2Parsed.majorVersion - v1Parsed.majorVersion;
+  }
+  // Finally compare minor version: v1beta2 > v1beta1
+  return v2Parsed.minorVersion - v1Parsed.minorVersion;
 };
 
-export const getLatestVersionForCRD = (crdVersions: CRDVersion[], legacyCRDVersion: string) => {
-  const sorted = crdVersionSort(crdVersions?.filter((version) => version.served));
-  return sorted?.[0]?.name || legacyCRDVersion;
+export const getLatestVersionForCRD = (crd: CustomResourceDefinitionKind) => {
+  const sorted = crd.spec.versions
```

Comment:
nit:
I think it's okay here because it's readable and we can probably assume the versions list will always be fairly short,  but I always like to point out where we are using chained iterative functions. This could cost performance if n became very large or this function became nested within other iterative functions.